### PR TITLE
implementing Fluent OS and OSProfile chain in VM

### PR DESCRIPTION
**Pull request by anuchandy**
https://github.com/Azure/azure-sdk-for-java/issues/697

### CODE REVIEW

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java [hunk 1]
```diff
@@ -98,52 +99,235 @@
...
+         * @return The next stage of the Windows virtual machine definition.
+         */
+        DefinitionWithDataDisk withWindowsOSDisk(String osDiskUrl);
+
+        /**
+         * Specifies the specialized Linux OS disk to be attached to the virtual machine.
+         * @param osDiskUrl The url to the OS disk in the Azure Storage account.
+         * @return The next stage of the Linux virtual machine definition.
+         */
+        DefinitionWithDataDisk withLinxOSDisk(String osDiskUrl);
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: linux

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java [hunk 2]
```diff
@@ -98,52 +99,235 @@
...
+         * Specifies that VM Agent should be provisioned.
+         * @return The optional windows configurations
+         */
+        DefinitionWithWindowsConfiguration enableVMAgent();
+
+        /**
+         * Specifies that VM Agent should not be provisioned.
+         * @return The optional windows configurations
+         */
+        DefinitionWithWindowsConfiguration disableVMAgent();
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: maybe we can default all to disabled unless users call enable?

Comment (anuchandy) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: node.js CLI and portal enables VM agent by default, so thought of having same for java lib as well.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java (hunk 1 above)
Comment (anuchandy) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: good catch

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java [hunk 3]
```diff
@@ -98,52 +99,235 @@
...
+         * @return The next stage of the virtual machine definition.
+         */
+        DefinitionWithOSType withImage(String userImageUrl);
+
+        /**
+         * Specifies the specialized Windows OS disk to be attached to the virtual machine.
+         * @param osDiskUrl The url to the OS disk in the Azure Storage account.
+         * @return The next stage of the Windows virtual machine definition.
+         */
+        DefinitionWithDataDisk withWindowsOSDisk(String osDiskUrl);
```
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: Should this return `DefinitionWithDataDisk` or `DefintionWithNextTODO`?

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java (hunk 2 above)
Comment (jianghaolu) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: I see, then maybe we can remove `enable`s. As Windows recommends turning on AutoUpdate as well.

Comment (anuchandy) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: make sense.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java (hunk 3 above)
Comment (anuchandy) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: right, will correct it

**Comment by jianghaolu** [comment]
Updated here: https://github.com/Azure/azure-sdk-for-java/issues/697

**Pull request merged**